### PR TITLE
Hardening/117 xxx individual context entity attributes

### PR DESCRIPTION
**Pull request by kzangeli**
Removed the service routines for
GET/POST/PUT/DELETE /v1/contextEntityies/{entity::id}/attributes

Made them use the already existing service routines for
GET/POST/PUT/DELETE /v1/contextEntityies/{entity::id}

This change has provoked a few changes in the tests, where /v1/contextEntityies/{entity::id}/attributes is used, as now the stdops QueryContext/UpdateContext are used instead of the convenience map operations.

Two unit-tests for another convop (/v1/contextEntityies/{entity::id}/attribute/{attributeName}/{metaIdValue}) have had to be disabled, to be enabled again once that convop is converted to using std-ops.
[ Forgot to add that FIXME by the way - coming right away! ]

### CODE REVIEW

File: test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test [hunk 1]
```diff
@@ -120,7 +120,10 @@ Date: REGEX(.*)
                 "reasonPhrase": "OK"
             }
         }
-    ]
+    ], 
+    "id": "E1",
```
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test]: Side-effect bug fix?

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test [hunk 2]
```diff
@@ -160,12 +160,15 @@ Date: REGEX(.*)
...
 2: ++++++++++++++++++++
 HTTP/1.1 200 OK
-Content-Length: 685
+Content-Length: 755
 Content-Type: application/xml
 Date: REGEX(.*)
 
 <?xml version="1.0"?>
 <appendContextElementResponse>
+  <entityId type="" isPattern="false">
```
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: Why this has changed?

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test [hunk 3]
```diff
@@ -211,7 +214,7 @@ Date: REGEX(.*)
   <contextAttributeList>
     <contextAttribute>
       <name>A1</name>
-      <type/>
+      <type>Room</type>
```
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: Maybe another case of #585 ?

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test [hunk 4]
```diff
@@ -44,7 +44,6 @@ Date: REGEX(.*)
     <entityId type="" isPattern="false">
       <id>entity01</id>
     </entityId>
-    <contextAttributeList/>
```
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test]: Another "funny" change... This is completly harmless but... why?

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test [hunk 5]
```diff
@@ -75,31 +75,29 @@ orionCurl --url "$url" --payload "${payload}" -X "PUT"
...
-    <contextAttributeResponse>
-      <contextAttributeList/>
-      <statusCode>
-        <code>404</code>
-        <reasonPhrase>No context element found</reasonPhrase>
-        <details>entity01</details>
-      </statusCode>
-    </contextAttributeResponse>
-  </contextResponseList>
+  <errorCode>
```
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test]: Side effect bug fix?

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: Well, we decided we wanted to add the info of the entity for AppendContextElementResponse when used ... (I don't remember exactly why) ... with GET /v1/contextEntities ... No, no me acuerdo

File: test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test (hunk 1 above)
Comment (kzangeli) [inline test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test]: Well, it is simply the adding of entity in the response

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: Yes, definitely
Already covered in CHANGES_NEXT_RELEASE with this line:
- Fix: A few more of the 'responses without entityId::type filled in' found and fixed (Issue #585)

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test]: Yeah, this was another improvement we got for free.
Different render method - my guess.
I haven't really spent any time in trying to understand these changes, as long as I've seen them as OK.

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test (hunk 5 above)
Comment (kzangeli) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test]: This was a fix I had to do a few days ago, for the 'without /attribute'. Here it came for free.
Now, if we want it for 'without' we probably want it for 'with' also.

File: test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test (hunk 1 above)
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_compound_values/convop_attributes_as_json_objects.test]: NTC

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: I think is again another side-effect of have now "/attribute" exactly equal to the other operation.

NTC

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test (hunk 4 above)
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_get_fails.test]: NTC

File: test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test (hunk 5 above)
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/individual_context_entity_attributes_put_create.test]: NTC

File: test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/000_ngsi10_convenience/attribute_value_instance.test]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**